dali/operators/decoder/inflate/inflate_params.h

```diff
@@ -0,0 +1,388 @@
+      if (num_chunks == 0) {
+        continue;
+      }
+      auto in_chunk_size = in_shape[sample_idx].num_elements();
+      for (int chunk_idx = 0; chunk_idx < num_chunks; chunk_idx++, flat_chunk_idx++) {
+        bool is_last_chunk = chunk_idx == num_chunks - 1;
+        int64_t next_chunk_offset = is_last_chunk ? in_chunk_size : offsets_[flat_chunk_idx + 1];
+        int64_t chunk_size = next_chunk_offset - offsets_[flat_chunk_idx];
+        assert(!is_last_chunk || chunk_size > 0);  // by offsets validation with `ValidateChunks`
+        DALI_ENFORCE(chunk_size > 0, non_monotone_error(sample_idx));
+        sizes_.push_back(chunk_size);
+      }
+    }
+    assert(offsets_.size() == sizes_.size());
+  }
+
+  template <int ndim>
+  void InferOffsetsFromSizes(const TensorListShape<ndim> &in_shape) {
+    assert(sizes_.size() == chunks_per_sample_.num_elements());
+    auto batch_size = in_shape.num_samples();
+    offsets_.clear();
+    offsets_.reserve(chunks_per_sample_.num_elements());
+    for (int64_t sample_idx = 0, flat_chunk_idx = 0; sample_idx < batch_size; sample_idx++) {
+      auto num_chunks = chunks_per_sample_[sample_idx].num_elements();
+      if (num_chunks == 0) {
+        continue;
+      }
+      int64_t cum_offset = 0;
+      auto sample_num_bytes = in_shape[sample_idx].num_elements();
+      for (int chunk_idx = 0; chunk_idx < num_chunks; chunk_idx++, flat_chunk_idx++) {
+        offsets_.push_back(cum_offset);
+        cum_offset += sizes_[flat_chunk_idx];
+      }
+      DALI_ENFORCE(cum_offset <= sample_num_bytes,
+                   make_string("The sum of chunk sizes for sample of idx ", sample_idx,
+                               " exceeds the total size of the sample."));
+    }
+    assert(offsets_.size() == sizes_.size());
+  }
+
+  void SetupOutputShape(const TensorListView<StorageCPU, const int> &provided_shape) {
+    DALI_ENFORCE(provided_shape.sample_dim() == 0 || provided_shape.sample_dim() == 1,
+                 make_string("The shape argument must be a scalar or a 1D tensor, got tensor with ",
+                             provided_shape.sample_dim(), " extents."));
+    auto chunk_shapes = ParseOutputShape(provided_shape);
+    if (!HasChunks()) {
+      output_shape_ = chunk_shapes;
+    } else {
+      int batch_size = chunk_shapes.num_samples();
+      output_shape_.resize(batch_size, chunk_shapes.sample_dim() + 1);
+      for (int sample_idx = 0; sample_idx < batch_size; sample_idx++) {
+        int num_chunks = chunks_per_sample_[sample_idx].num_elements();
+        output_shape_.set_tensor_shape(
+            sample_idx, shape_cat(TensorShape<>{num_chunks}, chunk_shapes[sample_idx]));
+      }
+    }
+  }
+
+  TensorListShape<> ParseOutputShape(const TensorListView<StorageCPU, const int> &provided_shape) {
+    auto num_samples = provided_shape.num_samples();
+    if (num_samples == 0) {
+      return {};
+    }
+    int sample_dim = provided_shape[0].num_elements();
```

Comment:
Shouldn't you verify that there's at least one element?

Reply:
Just for completeness (because it is silly example to use compression), allowing for num_elements = 0 covers compressed scalars.